File: node/examples/recognize.ts
// Run: npx tsx examples/recognize.ts [path to wave file]
import 'dotenv/config'

import { SttClient } from '@aristech-org/stt-client'
import fs from 'fs'
import path from 'path'

// Use the second command line argument as the path to the wave file
let fileName = process.argv[2]
if (!fileName) {
  // Fallback to test.wav in the repository root
  const scriptDir = path.dirname(new URL(import.meta.url).pathname)
  fileName = path.join(scriptDir, '../../test.wav')
} else if (!fs.existsSync(fileName) || !fileName.endsWith('.wav')) {
  console.error(`The file ${fileName} does not exist or does not end with .wav`)
  process.exit(1)
}

const client = new SttClient()
const results = await client.recognizeFile(fileName, { specification: { model: process.env.MODEL } })
console.log(results.map(r => r.chunks[0].alternatives[0].text).join('\n'))

File: python/examples/recognize.py
import sys
import os
from aristech_stt_client import SttClient, RecognitionConfig, RecognitionSpec
from dotenv import load_dotenv

load_dotenv()
model = os.getenv("MODEL")

# Get the repository root
repo_root = os.path.dirname(os.path.dirname(os.path.abspath(__file__ + "/..")))

client = SttClient()

# Load test.wav from the repository root or the first argument
# We do not have to specify the sample rate here because this is figured out by the client when using the recognize_file method
file_path = sys.argv[1] if len(sys.argv) > 1 else os.path.join(repo_root, "test.wav")
results = client.recognize_file(file_path, RecognitionConfig(specification=RecognitionSpec(model=model)))
print('\n'.join([r.chunks[0].alternatives[0].text for r in results]))